packages/apollo-fetch-upload/src/index.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  createApolloFetch,
+  constructDefaultOptions,
+  // Types:
+  ApolloFetch,
+  GraphQLRequest,
+} from 'apollo-fetch';
+import { extractFiles } from 'extract-files';
+
+export function constructUploadOptions(
+  requestOrRequests: GraphQLRequest | GraphQLRequest[],
+  options: RequestInit,
+): RequestInit {
+  const files = extractFiles(requestOrRequests);
```

Comment:
@jaydenseric currently, `reqeustOrRequests` isn't used after it is incorporated into the fetch options, so should be safe to modify in `constructOptions`/we don't need a defensive copy in `apollo-fetch`.

In general from `apollo-fetch`'s perspective, once we have called an external function, such as `constructOptions`, the references passed as arguments should not be used by `apollo-fetch`